k4MarlinWrapper/k4MarlinWrapper/converters/EDM4hep2Lcio.h

```diff
@@ -68,9 +68,11 @@ class EDM4hep2LcioTool : public AlgTool, virtual public IEDMConverter {
   // EventDataSvc that is used together with IOSvc
   ServiceHandle<IDataProviderSvc> m_eventDataSvc;
   // Metadata service from k4FWCore that is used together with IOSvc
-  SmartIF<IMetadataSvc>           m_metadataSvc;
-  std::vector<std::string>        m_collectionNames;
-  std::map<uint32_t, std::string> m_idToName;
+  SmartIF<IMetadataSvc> m_metadataSvc;
+  /// A (caching) map of original to new collection names that will be populated
+  /// during the first conversion
+  std::unordered_map<std::string, std::string> m_collsToConvert{};
```

Comment:
```suggestion
  std::map<std::string, std::string> m_collsToConvert{};
```
Maybe not worth it to be an unordered_map since the number of collections is probably never going to be very big?